examples/plot_vector_testing.py

```diff
@@ -0,0 +1,158 @@
+"""
+Testing Vector Plots Created Using Geopandas with MatPlotCheck
+==============================================================
+
+This vignette will show you how to use Matplotcheck to test spatial vector data
+plots created using Geopandas.
+
+"""
+
+################################################################################
+# Setup
+# -----
+# To begin, import the libraries that you need.
+
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+from shapely.geometry import Polygon, LineString
+import geopandas as gpd
+from matplotcheck.vector import VectorTester
+
+# Create geometry objects
+
+# Create a polygon GeoDataFrame
+coords = [(2, 4), (2, 4.25), (4.25, 4.25), (4.25, 2), (2, 2)]
+coords_b = [(i[0]+5, i[1]+7) for i in coords]
+polygon_a = Polygon(coords)
+polygon_b = Polygon(coords_b)
+polygon_gdf = gpd.GeoDataFrame(
+    [1, 2], geometry=[polygon_a, polygon_b], crs="epsg:4326")
+
+
+# Create a line GeoDataFrame
+linea = LineString([(1, 1), (2, 2), (3, 2), (5, 3)])
+lineb = LineString([(3, 4), (5, 7), (12, 2), (10, 5), (9, 7.5)])
+line_gdf = gpd.GeoDataFrame([1, 2], geometry=[linea, lineb], crs="epsg:4326")
+
+# Create a multiline GeoDataFrame
+linec = LineString([(2, 1), (3, 1), (4, 1), (5, 2)])
+multi_line_gdf = gpd.GeoDataFrame(geometry=gpd.GeoSeries(
+    [line_gdf.unary_union, linec]), crs="epsg:4326")
+multi_line_gdf["attr"] = ["road", "stream"]
+
+# Create a point GeoDataFrame
+
+points = pd.DataFrame(
+    {
+        "lat": np.array([1, 2, 1, 0, 4]),
+        "lon": np.array([3, 4, 0, 0, 1]),
+    }
+)
+
+point_gdf = gpd.GeoDataFrame(
+    {"A": np.arange(5), "B": np.arange(5)}, geometry=gpd.points_from_xy(points.lon, points.lat), crs="epsg:4326"
+)
+point_gdf["attr2"] = ["Tree", "Tree", "Bush", "Bush", "Bush"]
+
+# Create symbology dictionary to use in the legend
+
+line_symb = {"road": "black", "stream": "blue"}
+point_symb = {"Tree": "green", "Bush": "brown"}
+
+################################################################################
+# Create Your Spatial Plot
+# -----------------------------------------------------------------
+# Above you created several GeoPandas GeoDataFrame objects that you want
+# to plot. To plot these data according to attribute value, you can group
+# the geometry by attributes and plot within a loop. Once you've created
+# your plot, you are ready to test it using MatPlotCheck.
+
+# Plot your data
+fig, ax = plt.subplots()
+polygon_gdf.plot(ax=ax, color="purple")
+
+# Plot your data by attributes using groupby
+for ctype, lines in multi_line_gdf.groupby('attr'):
+    color = line_symb[ctype]
+    label = ctype
+    lines.plot(color=color, ax=ax, label=label)
+
+size = 0
```

Comment:
@nkorinek this example is a bit odd. you are manually creating a size object. it would be better if the polygon GDF have a column attribute that was numeric. then you can scale by a specific column value rather than an arbitrary value. this is a more common use case for marker size. One example in our week 5 homework was the plot tree height size
we scaled markers by average tree height.
another example could be precipitation values or census population values, or income. just something a bit more meaningful is better

Reply:
I'll implement this and ping you when I have it ready!